stdpopsim/species.py

```diff
@@ -170,42 +182,112 @@ def get_contig(self, chromosome, genetic_map=None, length_multiplier=1):
             same chromosome-specific mutation and recombination rates.
             This option cannot currently be used in conjunction with the
             ``genetic_map`` argument.
+        :param inlusion_mask: If specified, ...
+        :param exclusion_mask: If specified, ...
+        :param float length: Used with a "generic" contig, specifies the
+            length of genome sequence for this contig. For a generic contig, mutation
+            and recombination rates are equal to the genome-wide average across all
+            autosomal chromosomes.
         :rtype: :class:`.Contig`
         :return: A :class:`.Contig` describing the section of the genome.
         """
         # TODO: add non-autosomal support
-        if (chromosome is not None and
-                chromosome.lower() in ("x", "y", "m", "mt", "chrx", "chry", "chrm")):
-            warnings.warn(stdpopsim.NonAutosomalWarning(
+        if chromosome is not None and chromosome.lower() in (
+            "x",
+            "y",
+            "m",
+            "mt",
+            "chrx",
+            "chry",
+            "chrm",
+        ):
+            warnings.warn(
+                stdpopsim.NonAutosomalWarning(
                     "Non-autosomal simulations are not yet supported. See "
                     "https://github.com/popsim-consortium/stdpopsim/issues/383 and "
-                    "https://github.com/popsim-consortium/stdpopsim/issues/406"))
-        chrom = self.genome.get_chromosome(chromosome)
-        if genetic_map is None:
-            logger.debug(f"Making flat chromosome {length_multiplier} * {chrom.id}")
-            gm = None
-            recomb_map = msprime.RecombinationMap.uniform_map(
-                chrom.length * length_multiplier, chrom.recombination_rate)
-        else:
+                    "https://github.com/popsim-consortium/stdpopsim/issues/406"
+                )
+            )
+        if chromosome is None:
+            if genetic_map is not None:
+                raise ValueError("Cannot use genetic map with generic contic")
             if length_multiplier != 1:
-                raise ValueError("Cannot use length multiplier with empirical maps")
-            logger.debug(f"Getting map for {chrom.id} from {genetic_map}")
-            gm = self.get_genetic_map(genetic_map)
-            recomb_map = gm.get_chromosome_map(chrom.id)
-
-        ret = stdpopsim.Contig(
-            recombination_map=recomb_map, mutation_rate=chrom.mutation_rate,
-            genetic_map=gm)
+                raise ValueError("Cannot use length multiplier for generic contig")
+            if inclusion_mask is not None or exclusion_mask is not None:
+                raise ValueError("Cannot use mask with generic contig")
+            if length is None:
+                raise ValueError("Must specify sequence_length of generic contig")
+            non_autosomal_lower = ["x", "y", "m", "mt", "chrx", "chry", "chrm"]
```

Comment:
Could you move this `non_autosomal_lower` up and use in the predicate for the `NonAutosomalWarning`? Not sure if you aleady did this before, but the change has been lost.